### PR TITLE
initialize all formData in next/Stepper

### PR DESCRIPTION
**Pull request by dagda1**
Signed-off-by: Paul Cowan <paul.cowan@cutting.scot>

## Hey, I just made a Pull Request!

I ran into a problem using the `next/Stepper` where `formData` fields with undefined values, e.g. for a step like this:

```json
schema: {
  properties: {
    firstName: {
      type: 'string',
    },
  },
},
```
Then the `formData` would be `{}` until a value was added.

The `validate` function that is returned from `createAcyncValidators` uses `Object.entries` on the `formData` to call each custom extension's validator function.

If no key exists in `formData`, then `validator` will not be called.

This solution initializes the `formData` fields by calling `getDefaultFormState` from `@rjsf/utils`.
      

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/early-parrots-guess.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-scaffolder': patch
+---
+
+initialize all formData in next/Stepper
```
Comment (benjdlambert) [inline .changeset/early-parrots-guess.md:5]:
```suggestion
Initialize all `formData` in the `Stepper` in `/next`
```

**Review by benjdlambert — COMMENTED** [review]
Just one comment about the changeset but this makes sense!

**Comment by camilaibs** [comment]
Hi @dagda1, thanks again for contributing, could you take a look at [this](https://github.com/backstage/backstage/pull/14660#discussion_r1025279818) comment and resolve the pull request conflicts when you have time? Please let me know if you can't keep this up this week so I can help 🙂

**Comment by dagda1** [comment]
@camilaibs I totally missed the suggestion, thank you.   Should be all good now

**Comment by benjdlambert** [comment]
@dagda1 looks like it's still the old changeset message ☹️

**Comment by dagda1** [comment]
> @dagda1 looks like it's still the old changeset message 😅

My bad, that should be it now.

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**